include/ensmallen_bits/cmaes/pop_cmaes_impl.hpp

```diff
@@ -0,0 +1,168 @@
+ * Strategy With Increasing Population Size" and BIPOP Covariance Matrix
+ * Adaptation Evolution Strategy as proposed by N. Hansen in "Benchmarking 
+ * a BI-population CMA-ES on the BBOB-2009 function testbed".
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_CMAES_POP_CMAES_IMPL_HPP
+#define ENSMALLEN_CMAES_POP_CMAES_IMPL_HPP
+
+#include "pop_cmaes.hpp"
+#include <ensmallen_bits/function.hpp>
+#include <random>
+
+namespace ens {
+
+template<typename SelectionPolicyType, typename TransformationPolicyType, bool UseBIPOPFlag>
+POP_CMAES<SelectionPolicyType,
+          TransformationPolicyType,
+          UseBIPOPFlag>::POP_CMAES(
+    const size_t lambda,
+    const TransformationPolicyType& transformationPolicy,
+    const size_t batchSize,
+    const size_t maxIterations,
+    const double tolerance,
+    const SelectionPolicyType& selectionPolicy,
+    double stepSize,
+    const double populationFactor,
+    const size_t maxRestarts,
+    const size_t maxFunctionEvaluations) :
+    CMAES<SelectionPolicyType, TransformationPolicyType>(
+        lambda, transformationPolicy, batchSize, maxIterations,
+        tolerance, selectionPolicy, stepSize),
+    populationFactor(populationFactor),
+    maxRestarts(maxRestarts),
+    maxFunctionEvaluations(maxFunctionEvaluations)
+{ /* Nothing to do. */ }
+
+template<typename SelectionPolicyType,
+         typename TransformationPolicyType,
+         bool UseBIPOPFlag>
+template<typename SeparableFunctionType, typename MatType, typename... CallbackTypes>
+typename MatType::elem_type POP_CMAES<SelectionPolicyType, 
+    TransformationPolicyType, UseBIPOPFlag>::Optimize(
+    SeparableFunctionType& function,
+    MatType& iterateIn,
+    CallbackTypes&&... callbacks)
+{
+  // Convenience typedefs.
+  typedef typename MatType::elem_type ElemType;
+
+  StoreBestCoordinates<MatType> sbc;
+  StoreBestCoordinates<MatType> overallSBC;
+  size_t totalFunctionEvaluations = 0;
+  size_t largePopulationBudget = 0;
+  size_t smallPopulationBudget = 0;
+
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_real_distribution<> dis(0.0, 1.0);
+
+  // First single run with default population size
+  MatType iterate = iterateIn;
+  ElemType overallObjective = CMAES<SelectionPolicyType, 
+      TransformationPolicyType>::Optimize(function, iterate, sbc, 
+                                          callbacks...);
+
+  overallSBC = sbc;
+  ElemType objective;
+  size_t evaluations;
+
+  size_t defaultLambda = this->PopulationSize();
+  size_t currentLargeLambda = defaultLambda;
+
+  double stepSizeDefault = this->StepSize();
+
+  // Print out the default population size
+  Info << "Default population size: " << defaultLambda << std::endl;
```

Comment:
```suggestion
  Info << "Default population size: " << defaultLambda << "." << std::endl;
```
I know this is super pedantic.  A couple similar comments below.